test/utils/server.js

```diff
@@ -1,73 +1,34 @@
-const yaml = require('js-yaml');
 const extend = require('extend');
+const logStream = require('./logStream');
+const TestRunner = require('service-runner/test/TestServer');
 
+class TestServiceTemplateNode extends TestRunner {
+    constructor(configPath = `${__dirname}/../../config.yaml`) {
+        super(configPath);
+    }
 
-// set up the configuration
-let config = {
-    conf: yaml.safeLoad(fs.readFileSync(`${__dirname}/../../config.yaml`))
-};
-// build the API endpoint URI by supposing the actual service
-// is the last one in the 'services' list in the config file
-const myServiceIdx = config.conf.services.length - 1;
-const myService = config.conf.services[myServiceIdx];
-config.uri = `http://localhost:${myService.conf.port}/`;
-config.service = myService;
-// no forking, run just one process when testing
-config.conf.num_workers = 0;
-// have a separate, in-memory logger only
-config.conf.logging = {
-    name: 'test-log',
-    level: 'trace',
-    stream: logStream()
-};
-// make a deep copy of it for later reference
-const origConfig = extend(true, {}, config);
-
-module.exports.stop = () => { return BBPromise.resolve(); };
-let options = null;
-const runner = new ServiceRunner();
-
-
-function start(_options) {
-
-    _options = _options || {};
-
-    if (!assert.isDeepEqual(options, _options)) {
-        console.log('starting test server'); // eslint-disable-line no-console
-        return module.exports.stop().then(() => {
-            options = _options;
-            // set up the config
-            config = extend(true, {}, origConfig);
-            extend(true, config.conf.services[myServiceIdx].conf, options);
-            return runner.start(config.conf)
-            .then((serviceReturns) => {
-                module.exports.stop = () => {
-                    console.log('stopping test server'); // eslint-disable-line no-console
-                    serviceReturns.forEach(servers =>
-                        servers.forEach(server =>
-                            server.shutdown()));
-                    return runner.stop().then(() => {
-                        module.exports.stop = () => {
-                            return BBPromise.resolve();
-                        };
-                    });
-                };
-                return true;
-            });
-        });
-    } else {
-        return BBPromise.resolve();
+    get config() {
+        if (!this._running) {
+            throw new Error('Accessing test service config before starting the service');
+        }
+
+        // build the API endpoint URI by supposing the actual service
+        // is the last one in the 'services' list in the config file
+        const myServiceIdx = this._runner._impl.config.services.length - 1;
+        const myService = this._runner._impl.config.services[myServiceIdx];
+        const uri = `http://localhost:${myService.conf.port}/`;
+        const service = myService;
+
+        let config = {
```

Comment:
This is a longer story. The original test server created a config. Then created a deep copy of the original config that was then added to the config itself. After a last minute sweep, I realized that some of the config fields are never used included the copy of itself. I used let to attach the copy. When I removed the superfluous fields I left the rest of the code intact. You are correct on both should be returned directly since manipulations are no longer needed

Reply:
Do I need a second ticket? I am already working on this project. Might as well change this locally and run it.